Sources/ParseSwift/Coding/ParseEncoder.swift

```diff
@@ -0,0 +1,259 @@
+        _ParseEncoderSingleValueEncodingContainer(
+            codingPath: codingPath,
+            dictionary: dictionary,
+            skippingKeys: skippedKeys
+        )
+    }
+
+    func unkeyedContainer() -> UnkeyedEncodingContainer {
+        _ParseEncoderUnkeyedEncodingContainer(
+            codingPath: codingPath,
+            dictionary: dictionary,
+            skippingKeys: skippedKeys
+        )
+    }
+
+    static func encode<T: Encodable>(
+        _ value: T,
+        with codingPath: [CodingKey],
+        skippingKeys skippedKeys: Set<String>
+    ) throws -> Any {
+        switch value {
+        case is Bool, is Int, is Int8, is Int16, is Int32, is Int64, is UInt, is UInt8, is UInt16, is UInt32, is UInt64,
+             is Float, is Double, is String, is Date:
+            return value
+        default:
+            let dictionary = NSMutableDictionary()
+            let encoder = _ParseEncoder(codingPath: codingPath, dictionary: dictionary, skippingKeys: skippedKeys)
+            try value.encode(to: encoder)
+
+            return codingPath.last.map { dictionary[$0.stringValue] ?? dictionary } ?? dictionary
+        }
+    }
+}
+
+// MARK: _ParseEncoderKeyedEncodingContainer
+internal struct _ParseEncoderKeyedEncodingContainer<Key: CodingKey>: KeyedEncodingContainerProtocol {
+    let codingPath: [CodingKey]
+    let dictionary: NSMutableDictionary
+    let skippedKeys: Set<String>
+
+    init(codingPath: [CodingKey], dictionary: NSMutableDictionary, skippingKeys: Set<String>) {
+        self.codingPath = codingPath
+        self.dictionary = dictionary
+        self.skippedKeys = skippingKeys
+    }
+
+    mutating func encodeNil(forKey key: Key) throws {
+        if skippedKeys.contains(key.stringValue) { return }
+
+        dictionary[key.stringValue] = nil
+    }
+
+    mutating func encode<T>(_ value: T, forKey key: Key) throws where T: Encodable {
+        if skippedKeys.contains(key.stringValue) { return }
+
+        dictionary[key.stringValue] = try _ParseEncoder.encode(
+            value,
+            with: codingPath + [key],
+            skippingKeys: skippedKeys
+        )
+    }
+
+    mutating func nestedContainer<NestedKey>(
+        keyedBy keyType: NestedKey.Type,
+        forKey key: Key
+    ) -> KeyedEncodingContainer<NestedKey> where NestedKey: CodingKey {
+        let container = _ParseEncoderKeyedEncodingContainer<NestedKey>(
+            codingPath: codingPath + [key],
+            dictionary: dictionary,
+            skippingKeys: skippedKeys
+        )
+
+        return KeyedEncodingContainer(container)
+    }
+
+    mutating func nestedUnkeyedContainer(forKey key: Key) -> UnkeyedEncodingContainer {
+        _ParseEncoderUnkeyedEncodingContainer(
+            codingPath: codingPath + [key],
+            dictionary: dictionary,
+            skippingKeys: skippedKeys
+        )
+    }
+
+    mutating func superEncoder() -> Encoder {
+        fatalError()
```

Comment:
Maybe fill in the fatal error reasoning here? "Error, not allowed to call superEncoder". 

Are you able to mark these as `@available(*, unavailable)`?

Reply:
I can't do the `unavailable` thing because then the compiler complains about an unavailable method being used to satisfy protocol requirements. I'll add error messages though.